src/ruby/puppet-server-lib/puppet/server/master.rb

```diff
@@ -126,14 +89,17 @@ def terminate
   private
 
   def self.getManifests(env)
-    manifests = []
-    if env.manifest != Puppet::Node::Environment::NO_MANIFEST
-      if File.directory?(env.manifest)
-        manifests = Dir.glob(File.join(env.manifest, '**/*.pp')).sort
-      else
-        manifests = [env.manifest]
+    manifests =
+      case
+        when env.manifest == Puppet::Node::Environment::NO_MANIFEST
+          []
+        when File.directory?(env.manifest)
+          Dir.glob(File.join(env.manifest, '**/*.pp'))
```

Comment:
I read both Henrik's latest spec -- https://github.com/puppetlabs/puppet-specifications/blob/c6ce99fa77d92ac0db159befa86b94b209721fd4/language/modus-operandi.md#auto-loading-details -- and the current code in the "environment.rb" class -- https://github.com/puppetlabs/puppet/blob/4.3.2/lib/puppet/node/environment.rb#L485-L500 -- as not being more restrictive than this around the individual .pp files in the subpath of the `manifest` setting.  There are restrictions on the name of module directories under an environment's `modulepath` - only alphanumeric characters - but that all is encapsulated behind the `Module.all_manifests` method which is called below for modules.  Apparently, we don't exercise the same restrictions for subdirectories under the `manifest` setting.

Ideally, it would have been nice to be able to call some sort of method on the `Environment` class in Ruby that would provide back this list of files like `Module.all_manifests`does, but nothing appears to exist for that.  If we get back around to PUP-5011 later and decided that we wanted to continue relying upon core Ruby Puppet to help with abstracting the methods for determining the files to be parsed, maybe adding a similar `all_manifests` for the `Environment` class would be useful - although I gathered that we hoped to eventually go the other way and do all of the manifest enumeration in Clojure code instead for performance.